exist-core/src/main/java/org/exist/xquery/CastExpression.java

```diff
@@ -190,7 +190,7 @@ public Function toFunction() throws XPathException {
 	    try {
             final QName qname = QName.parse(context, typeName);
             final FunctionSignature signature = new FunctionSignature(qname);
-            final SequenceType argType = new SequenceType(Type.ITEM, Cardinality.EXACTLY_ONE);
+            final SequenceType argType = new SequenceType(Type.ITEM, Cardinality.ZERO_OR_ONE);
```

Comment:
I think that the specification: [18.1 Constructor functions for XML Schema built-in atomic types](https://www.w3.org/TR/xpath-functions-31/#constructor-functions-for-xsd-types) says that this should be `Type.ATOMIC` and not `Type.ITEM`. Otherwise, LGTM.

Reply:
Both Saxon and basex allow `item()?` to be passed to cast constructor functions.
```
xs:integer([1]), 
xs:string(<foo>test</foo>)
``` 
are both valid

Reply:
I asked for clarification on XML Slack https://xmlcom.slack.com/archives/C011NLXE4DU/p1688742869580379?thread_ts=1688552697.034799&cid=C011NLXE4DU

Reply:
OK, so atomisation will be done before the argument is passed in. Will change the type.